### DISCUSSION TITLE
How to put a .po file in a NuGet package

### DISCUSSION THREAD

**Discussion by andrew-klein-sage** [OrchardCMS/OrchardCore#10722] [Q&A]
I have an assortment of dotnet projects in separate github repositories that have a significant amount of common code that I want to put into a nuget package, but the common code uses IStringLocalizer for a couple of strings.  I'm using services.AddPortableObjectLocalization() in my Startup.cs and I'd like to not have to have the common code strings in the .po file for all the repositories.  How can I put a .po file into the nuget package and get properly localized?

Thanks!

**Comment by hishamco** [comment]
> How can I put a .po file into the nuget package and get properly localized?

OC already shipped varity of translations located in https://github.com/OrchardCMS/OrchardCore.Translations and hosted in NuGet. I'm not sure why you need to add a new .PO file?

**Reply by andrew-klein-sage** [reply]
I have my own strings to localize, in the form of:
`msgid "StringToLocalize"`
`msgstr "I have been localized"`

In the source of my nuget package I have something like:
`namespace MyNamespace {`
`    public class MyClass {`
`      private readonly IStringLocalizer<TenantValidator> _localizer;`
`      public MyClass(IStringLocalizer<MyClass> localizer) {_localizer = localizer;}`
`      public string GetLocalizedString() {return _localizer["StringToLocalize"];}`
`   }`
`}`

I know that in the consumer of my nuget package I need the call to `AddPortableObjectLocalization` but I don't know how to distribute the translations.  I've added the `PackageTranslationFiles` target to the csproj file for my nuget package but it didn't help.  It's possible I'm looking at this completely wrong...but hopefully the problem I want to solve is clear.

**Reply by hishamco** [reply]
If you have a custom module, you can do like what we did in OC

**Comment by ns8482e** [comment]
Include `.PO` as `content` in your project. I hope this will help - https://docs.microsoft.com/en-us/aspnet/core/fundamentals/portable-object-localization?view=aspnetcore-6.0
